docs/general-usage/middleware.rst

```diff
@@ -0,0 +1,71 @@
+Middleware
+==========
+
+You can use ``middleware`` to affect the evaluation of fields in your schema.
+
+A middleware is any object or function that responds to ``resolve(next_middleware, *args).``
+
+Inside that method, it should either:
+
+    - Send ``resolve`` to the next middleware to continue the evaluation; or
+    - Return a value to end the evaluation early.
+
+
+Resolve arguments
+-----------------
+
+Middlewares ``resolve`` is invoked with several arguments:
+
+- ``next`` represents the execution chain. Call ``next`` to continue evaluation.
+- ``root`` is the root value object passed throughout the query.
+- ``info`` is the resolver info.
+- ``args`` is the dict of arguments passed to the field.
+
```

Comment:
This is a copy of https://docs.graphene-python.org/en/latest/execution/middleware/.
Maybe condense it to the first part and link out to the Graphene docs?